### PR TITLE
Better type for click 'type' argument

### PR DESCRIPTION
**Pull request by euresti**
The call accepts tuples of types and callback methods

### CODE REVIEW

File: third_party/2and3/click/types.pyi [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from typing import Any, Callable, IO, Iterable, List, Optional, TypeVar, Union
+from typing import Any, Callable, IO, Iterable, List, Optional, TypeVar, Union, Tuple as PyTuple
```
Comment (JelleZijlstra) [inline third_party/2and3/click/types.pyi]: Should be `_PyTuple` so it's not exported.

File: third_party/2and3/click/types.pyi [hunk 2]
```diff
@@ -270,7 +270,9 @@ class UUIDParameterType(ParamType):
         ...
 
 
-def convert_type(ty: Any, default: Optional[Any] = ...) -> ParamType:
+_ConvertibleType = Union[type, ParamType, PyTuple[type, ...], Callable[[str], Any], Callable[[Optional[str]], Any]]
```
Comment (JelleZijlstra) [inline third_party/2and3/click/types.pyi]: Isn't `Callable[[str], Any]` redundant with `Callable[[Optional[str]], Any]`?

Comment (euresti) [reply third_party/2and3/click/types.pyi]: It doesn't seem to be.  

```
from typing import Union, Callable, Optional, Any

With = Union[None, Callable[[str], Any], Callable[[Optional[str]], Any]]
Without = Union[None, Callable[[Optional[str]], Any]]

def cb1(a: str) -> None:...
def cb2(a: Optional[str]) -> None: ...

def fn_with(cb: With) -> None: ...
def fn_without(cb: Without) -> None: ...

fn_with(cb1)
fn_with(cb2)

fn_without(cb1)   # Argument 1 to "fn_without" has incompatible type "Callable[[str], None]"; expected "Optional[Callable[[Optional[str]], Any]]"
fn_without(cb2)
```

**Comment by euresti** [comment]
Converted to _PyTuple.  But didn't change the other one because otherwise my uses of click.option fail with `--strict-optional`

**Pull request merged**